Framework/API/src/FileBackedExperimentInfo.cpp

```diff
@@ -3,8 +3,8 @@
 
 #include <sstream>
 
-#include <nexus/NeXusException.hpp>
 #include <nexus/NeXusFile.hpp>
+#include <nexus/NeXusException.hpp>
```

Comment:
Yes it does not compile if it is the otherway round (I don't know why). I believe in the cases you mentioned the `NeXusFile.hpp` must get included by some other header before.